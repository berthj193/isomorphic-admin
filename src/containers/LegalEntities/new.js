import React, { Component } from 'react';
import PropTypes from 'prop-types';

import LegalEntitiesForm from './legalEntitiesForm';

import withStoresManager from '../../components/apolloProviders/withStoresManager';

import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';

import { createLegalEntity, getList } from '../../graphql/legalEntities';

export class NewLegalEntity extends Component {
  static propTypes = {
    mutate: PropTypes.func.isRequired,
  };

  handleSubmit = values => {
    this.props.mutate({
      mutation: createLegalEntity,
      variables: {
        ...values,
      },
      refetchQueries: [{ query: getList }],
      messages: ['Legal entity added successfully'],
    });
  };

  render() {
    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Legal entities</h1>
          <ButtonList buttons={[{ link: '/dashboard/legal_entities', title: 'Go back' }]} />
          <LegalEntitiesForm
            onSubmit={this.handleSubmit}
          />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(NewLegalEntity);
