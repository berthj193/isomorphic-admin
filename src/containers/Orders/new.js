import React, { Component } from 'react';
import PropTypes from 'prop-types';

import OrderForm from './orderForm';

import withOrdersManager from '../../components/apolloProviders/withOrdersManager';

import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';

import { placeOrder } from '../../graphql/orders';

export class NewOrder extends Component {
  static propTypes = {
    mutate: PropTypes.func.isRequired,
  };

  handleSubmit = values => {
    const { shipping_address_id, shipping_rate_id, line_items } = values;

    this.props.mutate({
      mutation: placeOrder,
      variables: {
        ...values,
        shipping_address_id: parseInt(shipping_address_id, 0),
        shipping_rate_id: parseInt(shipping_rate_id, 0),
        line_items: JSON.stringify(line_items),
      },
      messaages: ['Order added successfully'],
    });
  };

  render() {
    const initialValues = {
      currency: 'GBP',
    };

    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Orders</h1>
          <ButtonList buttons={[{ link: '/dashboard/orders', title: 'Go back' }]} />
          <OrderForm
            onSubmit={this.handleSubmit}
            initialValues={initialValues}
          />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withOrdersManager(NewOrder);
