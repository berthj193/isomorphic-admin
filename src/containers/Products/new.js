import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ProductForm from './productForm';

import withStoresManager from '../../components/apolloProviders/withStoresManager';
import storeQueries from '../../graphql/stores';
import productQueries from '../../graphql/products';

import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';
import Error from '../../components/error/error';

export class NewProduct extends Component {
  static propTypes = {
    mutate: PropTypes.func.isRequired,
    stores: PropTypes.arrayOf(PropTypes.object),
    error: PropTypes.string,
  };

  handleSubmit = ({ default_name, default_description, default_master_sku }) =>
    this.props.mutate({
      mutation: productQueries.createProduct(),
      variables: {
        default_name,
        default_description,
        default_master_sku,
      },
      refetchQueries: [
        { query: productQueries.getProducts() },
      ],
      messages: ['Product created successfully'],
    });

  render() {
    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Product</h1>
          <ButtonList buttons={[{ link: '/dashboard/products', title: 'Go back' }]} />
          <ProductForm
            onSubmit={this.handleSubmit}
            stores={this.props.stores}
            resetOnSuccess
          />
          <Error error={this.props.error} />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(NewProduct, {
  query: storeQueries.storesAutocomplete,
});
