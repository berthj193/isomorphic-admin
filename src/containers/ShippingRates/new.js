import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ShippingRateForm from './shippingRateForm';

import withStoresManager from '../../components/apolloProviders/withStoresManager';

import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';

import { createShippingRate, getList } from '../../graphql/shippingRates';

export class NewShippingRate extends Component {
  static propTypes = {
    mutate: PropTypes.func.isRequired,
  };

  handleSubmit = values => {
    this.props.mutate({
      mutation: createShippingRate,
      variables: {
        ...values,
        price_pence: parseInt(values.price_pence, 0),
        min_items: parseInt(values.min_items, 0),
        max_items: parseInt(values.max_items, 0),
      },
      refetchQueries: [
        { query: getList },
      ],
      messages: ['Shipping rate was added successfully'],
    });
  };

  render() {
    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Shipping rates</h1>
          <ButtonList buttons={[{ link: '/dashboard/shipping_rates', title: 'Go back' }]} />
          <ShippingRateForm
            onSubmit={this.handleSubmit}
            resetOnSuccess
          />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(NewShippingRate);
