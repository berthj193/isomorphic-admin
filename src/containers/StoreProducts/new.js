import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Form from './storeProductForm';

import withStoresManager from '../../components/apolloProviders/withStoresManager';

import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';

import storeProductQueries from '../../graphql/storeProducts';

export class NewStoreProduct extends Component {
  static propTypes = {
    mutate: PropTypes.func.isRequired,
  };

  handleSubmit = values => {
    this.props.mutate({
      mutation: storeProductQueries.createStoreProduct,
      variables: {
        ...values,
      },
      messages: ['Store product was added successfully'],
    });
  };

  render() {
    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Store Products</h1>
          <ButtonList buttons={[{ link: '/dashboard/store_products', title: 'Go back' }]} />
          <Form
            onSubmit={this.handleSubmit}
            initialValues={{
              use_defaults: true,
            }}
          />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(NewStoreProduct);
