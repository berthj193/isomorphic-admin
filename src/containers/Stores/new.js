import React, { Component } from 'react';
import PropTypes from 'prop-types';

import StoreForm from './storeForm';

import withStoresManager from '../../components/apolloProviders/withStoresManager';

import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';

import storeQueries from '../../graphql/stores';

class NewStore extends Component {
  static propTypes = {
    mutate: PropTypes.func,
  };

  handleSubmit = values => {
    const options = values.options.reduce((prev, curr) => ({
      ...prev,
      [curr.key]: curr.value,
    }), {});
    this.props.mutate({
      mutation: storeQueries.createStore,
      variables: {
        ...values,
        legal_entity_id: parseInt(values.legal_entity_id, 0),
        warehouse_id: parseInt(values.warehouse_id, 0),
        channel_id: parseInt(values.channel_id, 0),
        options: JSON.stringify(options),
      },
      refetchQueries: [{
        query: storeQueries.getList,
      }],
      messages: ['Store added successfully'],
    });
  };

  render() {
    const initialValues = {
      options: [
        {
          key: 'email_templates_order_completed',
          value: '1',
        },
        {
          key: 'email_templates_order_dispatched',
          value: '3',
        },
      ],
    };

    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Store</h1>
          <ButtonList buttons={[{ link: '/dashboard/stores', title: 'Go back' }]} />
          <StoreForm
            onSubmit={this.handleSubmit}
            initialValues={initialValues}
          />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(NewStore);
