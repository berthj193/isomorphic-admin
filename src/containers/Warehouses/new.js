import React, { Component } from 'react';
import PropTypes from 'prop-types';

import WarehouseForm from './warehouseForm';

import withStoresManager from '../../components/apolloProviders/withStoresManager';

import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';

import { createWarehouse, getList } from '../../graphql/warehouses';

export class NewWarehouse extends Component {
  static propTypes = {
    mutate: PropTypes.func.isRequired,
  };

  handleSubmit = values => {
    this.props.mutate({
      mutation: createWarehouse,
      variables: {
        ...values,
      },
      refetchQueries: [{
        query: getList,
      }],
      messages: ['Warehouse added successfully'],
    });
  };

  render() {
    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Warehouses</h1>
          <ButtonList buttons={[{ link: '/dashboard/warehouses', title: 'Go back' }]} />
          <WarehouseForm
            onSubmit={this.handleSubmit}
          />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(NewWarehouse);
