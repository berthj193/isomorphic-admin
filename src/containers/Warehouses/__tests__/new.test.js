import React from 'react';
import { NewWarehouse } from '../new';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<NewWarehouse />', () => {
  const getComponent = (props = { mutate: () => null }) => shallow(<NewWarehouse {...props} />);
  const mutate = jest.fn();

  beforeEach(() => {
    jest.resetAllMocks();
    mutate.mockReturnValue(Promise.resolve({ foo: 'bar' }));
  });

  it('should render properly', () => {
    expect(toJson(getComponent())).toMatchSnapshot();
  });

  it('should execute mutate function on submit', () => {
    const wrapper = getComponent({ mutate });
    wrapper.instance().handleSubmit({});
    expect(mutate).toBeCalled();
  });
});
