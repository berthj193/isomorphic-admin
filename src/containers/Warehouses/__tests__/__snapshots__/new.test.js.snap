// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewWarehouse /> should render properly 1`] = `
<LayoutWrapper
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <LayoutContent>
    <h1>
      Warehouses
    </h1>
    <ButtonList
      buttons={
        Array [
          Object {
            "link": "/dashboard/warehouses",
            "title": "Go back",
          },
        ]
      }
    />
    <ReduxFormWrapper
      onSubmit={[Function]}
    />
  </LayoutContent>
</LayoutWrapper>
`;
